mathics/builtin/logic.py

```diff
@@ -116,9 +116,139 @@ class Not(PrefixOperator):
     rules = {
         'Not[True]': 'False',
         'Not[False]': 'True',
+        'Not[Not[expr_]]': 'expr',
     }
 
 
+class Implies(BinaryOperator):
+    """
+    <dl>
+    <dt>'Implies[$expr1$, $expr2$]'
+    <dt>'$expr1$ \u21D2 $expr2$'
+        <dd>evaluates each expression in turn, returning 'True'
+        as soon as the first expression evaluates to 'False'. If the
+        first expression evaluates to 'True', 'Implies' returns the
+        second expression.
+    </dl>
+
+    >> Implies[False, a]
+     = True
+    >> Implies[True, a]
+     = a
+
+    If an expression does not evaluate to 'True' or 'False', 'Implies'
+    returns a result in symbolic form:
+    >> Implies[a, Implies[b, Implies[True, c]]]
+     = a \u21D2 b \u21D2 c
+    """
+
+    operator = '\u21D2'
+    precedence = 200
+    attributes = ('HoldAll', 'OneIdentity')
+    grouping = 'Right'
+
+    def apply(self, x, y, evaluation):
+        'Implies[x_, y_]'
+
+        result0 = x.evaluate(evaluation)
+        if result0 == Symbol('False'):
+            return Symbol('True')
+        elif result0.is_true():
+            return y.evaluate(evaluation)
+        else:
+            return Expression('Implies', result0, y.evaluate(evaluation))
+
+
+class Equivalent(BinaryOperator):
+    """
+     <dl>
+     <dt>'Equivalent[$expr1$, $expr2$, ...]'
+     <dt>'$expr1$ \u21D4 $expr2$ \u21D4 ...'
+         <dd>is equivalent to
+         ($expr1$ && $expr2$ && ...) || (!$expr1$ && !$expr2$ && ...)
+     </dl>
+
+     >> Equivalent[True, True, False]
+      = False
+
+     If all expressions do not evaluate to 'True' or 'False', 'Equivalent'
+     returns a result in symbolic form:
+     >> Equivalent[a, b, c]
+      = a \u21D4 b \u21D4 c
+      Otherwise, 'Equivalent' returns a result in DNF
+      >> Equivalent[a, b, True, c]
+       = a && b && c
+     """
```

Comment:
can you also add a test for `Equivalent[]` (should be `True`) and `Equivalent[x]`, also `True`. Note that you can add hidden tests with `#>` (they won't show up in the docs).